packages/model-generator/src/s3_string_object_fetcher.ts

```diff
@@ -14,15 +15,27 @@ export class S3StringObjectFetcher {
    */
   fetch = async (uri: string) => {
     const { bucket, key } = this.parseS3Uri(uri);
-    const getSchemaCommandResult = await this.s3Client.send(
-      new GetObjectCommand({ Bucket: bucket, Key: key })
-    );
-    const schema = await getSchemaCommandResult.Body?.transformToString();
-    if (!schema) {
-      // eslint-disable-next-line amplify-backend-rules/prefer-amplify-errors
-      throw new Error('Error on parsing output schema');
+    try {
+      const getSchemaCommandResult = await this.s3Client.send(
+        new GetObjectCommand({ Bucket: bucket, Key: key })
+      );
+      const schema = await getSchemaCommandResult.Body?.transformToString();
+      if (!schema) {
+        // eslint-disable-next-line amplify-backend-rules/prefer-amplify-errors
+        throw new Error('Error on parsing output schema');
+      }
+      return schema;
+    } catch (caught) {
+      if (caught instanceof NoSuchBucket) {
+        throw new AmplifyUserError('BucketDoesNotExistError', {
+          message: `${caught.name} the bucket, ${bucket} does not exist.`,
+          resolution: `Ensure that ${bucket} exists and contains its expected contents before trying this command again.\n
+            To find the expected contents, create a new sandbox and investigate the contents of buckets with a similar name to this one.`,
+        });
+      } else {
```

Comment:
The `bucketName` or the `s3Uri` is not a user input but managed by amplify. It's saved in the stack outputs during the CFN deployment and then retrieved here https://github.com/aws-amplify/amplify-backend/blob/1400e6f65cf8e156b8aaeddc4cb37311a37a3fc3/packages/model-generator/src/create_graphql_models_generator.ts#L122 before passing to this method.

Error here would typically mean
1. Bug in the Amplify Backend
2. Some validations missing internally allowing a bad s3 URI to be passed here
3. Customer manually deleted this bucket.

the third is indeed a user error but how can we ascertain that the first two aren't an issue?

Reply:
That's the challenge, it's super hard with the current information available. My suggestion is to keep this as a `fault` for now and add more data in the error message that would help us and customers figure out what's the cause of this.

Reply:
I was able to replicate this by manually deleting buckets, but there could very well be a bug with these commands. Especially because the buckets you have to delete to trigger this error all have a `model-schema.graphql` file in them so users might be less likely to get rid of them because of the 2-step process of emptying then deleting the bucket.